pkg/api/render.go

```diff
@@ -59,22 +59,23 @@ func (hs *HTTPServer) RenderToPng(c *contextmodel.ReqContext) {
 	encoding := queryReader.Get("encoding", "")
 
 	result, err := hs.RenderService.Render(c.Req.Context(), rendering.RenderPNG, rendering.Opts{
-		TimeoutOpts: rendering.TimeoutOpts{
-			Timeout: time.Duration(timeout) * time.Second,
-		},
-		AuthOpts: rendering.AuthOpts{
-			OrgID:   c.SignedInUser.GetOrgID(),
-			UserID:  userID,
-			OrgRole: c.SignedInUser.GetOrgRole(),
+		CommonOpts: rendering.CommonOpts{
+			TimeoutOpts: rendering.TimeoutOpts{
+				Timeout: time.Duration(timeout) * time.Second,
+			},
+			AuthOpts: rendering.AuthOpts{
+				OrgID:   c.SignedInUser.GetOrgID(),
+				UserID:  userID,
+				OrgRole: c.SignedInUser.GetOrgRole(),
+			},
+			Path:            web.Params(c.Req)["*"] + queryParams,
+			Timezone:        queryReader.Get("tz", ""),
+			ConcurrentLimit: hs.Cfg.RendererConcurrentRequestLimit,
+			Headers:         headers,
 		},
 		Width:             width,
 		Height:            height,
-		Path:              web.Params(c.Req)["*"] + queryParams,
-		Timezone:          queryReader.Get("tz", ""),
-		Encoding:          encoding,
```

Comment:
This confuses me a bit. The function name is `RenderToPng`, and the `RenderService.Render` function is being called with a hardcoded type `rendering.RenderPNG`. That is why I removed it. If this could be called using 'pdf,' then we should think of changing names and confirm that changing the render type above on line 61 has no side effect.

Reply:
I've made the changes. Now the `renderType` is inferred by the encoding param and managed by the API consumer like before. WDYT? I don't want to change the behavior in this PR.